tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -123,31 +123,55 @@ def index_impl(self):
 
     return result
 
-  def scalars_impl(self, tag, run, output_format):
+  def scalars_impl(self, tag, run, experiment, output_format):
     """Result of the form `(body, mime_type)`."""
     if self._db_connection_provider:
       db = self._db_connection_provider()
       # We select for steps greater than -1 because the writer inserts
       # placeholder rows en masse. The check for step filters out those rows.
-      cursor = db.execute('''
-        SELECT
-          Tensors.step,
-          Tensors.computed_time,
-          Tensors.data,
-          Tensors.dtype
-        FROM Tensors
-        JOIN Tags
-          ON Tensors.series = Tags.tag_id
-        JOIN Runs
-          ON Tags.run_id = Runs.run_id
-        WHERE
-          Runs.run_name = ?
-          AND Tags.tag_name = ?
-          AND Tags.plugin_name = ?
-          AND Tensors.shape = ''
-          AND Tensors.step > -1
-        ORDER BY Tensors.step
-      ''', (run, tag, metadata.PLUGIN_NAME))
+      if experiment is None:
+        # TODO(stephanwlee): Deprecate this if-block when experiment_id becomes
```

Comment:
If the `IS` operator is used on `Runs.experiment_id` then it'll probably do what you want with one query. Our schema only defines fields as `NOT NULL` when it *must*, because required fields in SQL are similarly problematic to required fields in protobuf.

Reply:
I am sure I am misunderstanding you here. Sorry in advance.

What this code is trying to do is, in case one is NOT using the data-selector (flippable only via frontend flag), the experiment param is not passed and, in such case, we'd like to fallback to the previous behavior: returning all tensor that matches run name even across (sometimes, non-null) experiments. Is there a better way to conditionally provide the Where clause?

Reply:
If the user sets the `experiment_name` in the DB, then the user should need to enable experiment support in UI to see it. Otherwise, TB *must* ignore the data. It's necessary for forwards compatibility. These DBs are designed to be potentially TBs in size.